### PR TITLE
Add sanitized name cleanser to brokerquery and use it in summary and report grouping

### PR DESCRIPTION
**Pull request by jpalac**

### CODE REVIEW

File: src/ServiceControl.Transports.SQS/AmazonSQSQuery.cs [hunk 1]
```diff
@@ -292,4 +292,6 @@ public override string SanitizeEndpointName(string endpointName)
             return endpointName;
         }
     }
+
+    public override string SanitizedEndpointNameCleanser(string endpointName) => endpointName.ToLower();
```
Comment (johnsimons) [inline src/ServiceControl.Transports.SQS/AmazonSQSQuery.cs:296]: I checked SQS and the queue name is case sensitive.
```suggestion

```

Comment (jpalac) [reply src/ServiceControl.Transports.SQS/AmazonSQSQuery.cs]: removed it

File: src/ServiceControl.Transports/BrokerThroughput/IBrokerThroughputQuery.cs [hunk 2]
```diff
@@ -20,4 +20,5 @@ public interface IBrokerThroughputQuery
     KeyDescriptionPair[] Settings { get; }
     Task<(bool Success, List<string> Errors, string Diagnostics)> TestConnection(CancellationToken cancellationToken);
     string SanitizeEndpointName(string endpointName);
+    string SanitizedEndpointNameCleanser(string endpointName);
```
Comment (abparticular) [inline src/ServiceControl.Transports/BrokerThroughput/IBrokerThroughputQuery.cs:23]: Do we need another method? Is it not possible for `SanitizeEndpointName` to include any necessary "cleansing"?

File: src/Particular.LicensingComponent/ThroughputCollector.cs [hunk 3]
```diff
@@ -214,6 +214,13 @@ string Mask(string stringToMask)
         }
     }
 
+    string CleanseSanitizedName(string endpointName)
+    {
+        return throughputQuery == null ? defaultSanitizedEndpointNameCleanser(endpointName) : throughputQuery.SanitizedEndpointNameCleanser(endpointName);
+    }
+
+    Func<string, string> defaultSanitizedEndpointNameCleanser = endpointName => endpointName;
+
```
Comment (abparticular) [inline src/Particular.LicensingComponent/ThroughputCollector.cs:223]:
```suggestion
    string CleanseSanitizedName(string endpointName)
    {
        return throughputQuery == null ? endpointName : throughputQuery.SanitizedEndpointNameCleanser(endpointName);
    }
```

File: src/ServiceControl.Transports/BrokerThroughput/IBrokerThroughputQuery.cs (hunk 2 above)
Comment (jpalac) [reply src/ServiceControl.Transports/BrokerThroughput/IBrokerThroughputQuery.cs:23]: no as that's used for something else - to store the sanitized name. We are then using the cleansing to cleanse the audit and monitoring sanitized name to that it matches what the broker has.

Comment (johnsimons) [reply src/ServiceControl.Transports/BrokerThroughput/IBrokerThroughputQuery.cs:23]: yeah, i was thinking the same but i can see the reason

**Review by johnsimons — APPROVED** [review]

**Pull request merged**